src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/services/NotificationServiceTest.java

```diff
@@ -0,0 +1,138 @@
+import uk.gov.hmcts.reform.blobrouter.data.DbHelper;
+import uk.gov.hmcts.reform.blobrouter.data.envelopes.Envelope;
+import uk.gov.hmcts.reform.blobrouter.data.events.EnvelopeEvent;
+import uk.gov.hmcts.reform.blobrouter.data.events.EnvelopeEventRepository;
+import uk.gov.hmcts.reform.blobrouter.data.events.EventType;
+import uk.gov.hmcts.reform.blobrouter.data.rejectedenvelope.RejectedEnvelopeRepository;
+import uk.gov.hmcts.reform.blobrouter.servicebus.notifications.NotificationsPublisher;
+import uk.gov.hmcts.reform.blobrouter.servicebus.notifications.model.NotificationMsg;
+
+import java.util.List;
+import java.util.Optional;
+
+import static java.time.Instant.now;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static uk.gov.hmcts.reform.blobrouter.data.events.EventType.DELETED;
+
+@SpringBootTest
+@ExtendWith(MockitoExtension.class)
+@ActiveProfiles("db-test")
+class NotificationServiceTest {
+
+    @Autowired
+    private NotificationService notificationService;
+    @Autowired
+    private EnvelopeService envelopeService;
+    @Autowired
+    private RejectedEnvelopeRepository rejectedEnvelopeRepository;
+    @Autowired
+    private EnvelopeEventRepository envelopeEventRepository;
+    @Autowired
+    private DbHelper dbHelper;
+
+    @Captor
+    private ArgumentCaptor<NotificationMsg> notificationMsgCaptor;
+
+    @Mock
+    NotificationsPublisher notificationsPublisher;
+
+    @BeforeEach
+    void setUp() {
+        notificationService = new NotificationService(
+            notificationsPublisher,
+            rejectedEnvelopeRepository,
+            envelopeService
+        );
+    }
+
+    @AfterEach
+    void tearDown() {
+        dbHelper.deleteAll();
+    }
+
+    @Test
+    void should_send_notifications_for_the_rejected_envelopes() {
+        // given
+        // rejected envelope
+        var envelopeId1 = envelopeService.createNewEnvelope("bulkscan", "blob1.zip", now());
+        envelopeService.markAsRejected(envelopeId1, "duplicate file");
+
+        // envelope that is not rejected
+        var envelopeId2 = envelopeService.createNewEnvelope("bulkscan", "blob2.zip", now());
+        envelopeService.markAsDispatched(envelopeId2);
+
+        // rejected envelope
+        var envelopeId3 = envelopeService.createNewEnvelope("testcontainer", "blob3.zip", now());
+        envelopeService.markAsRejected(envelopeId3, "invalid signature");
+        envelopeService.saveEvent(envelopeId3, DELETED);
+
+        // when
+        notificationService.sendNotifications();
+
+        // then
+        verify(notificationsPublisher, times(2)).publish(any());
+
+        Optional<Envelope> envelope1 = envelopeService.findEnvelope(envelopeId1);
+        assertThat(envelope1).hasValueSatisfying(env -> assertThat(env.pendingNotification).isEqualTo(false));
```

Comment:
usually it's `assertThat(someBoolean).isFalse()`. there's another one on line 97